resource-aggregate/service/deleteDevices.go

```diff
@@ -0,0 +1,55 @@
+package service
+
+import (
+	"context"
+
+	"github.com/plgd-dev/cloud/pkg/log"
+	kitNetGrpc "github.com/plgd-dev/cloud/pkg/net/grpc"
+	"github.com/plgd-dev/cloud/resource-aggregate/commands"
+	"github.com/plgd-dev/cloud/resource-aggregate/cqrs/eventstore"
+	"github.com/plgd-dev/kit/strings"
+	"google.golang.org/grpc/codes"
+)
+
+func getUniqueDeviceIdsFromDeleteRequest(request *commands.DeleteDevicesRequest) []string {
+	deviceIds := make(strings.Set)
+	for _, deviceId := range request.DeviceIds {
+		if deviceId != "" {
+			deviceIds.Add(deviceId)
+		}
+	}
+	return deviceIds.ToSlice()
+}
+
+func (r RequestHandler) DeleteDevices(ctx context.Context, request *commands.DeleteDevicesRequest) (*commands.DeleteDevicesResponse, error) {
+	deviceIds := getUniqueDeviceIdsFromDeleteRequest(request)
+	if len(deviceIds) == 0 {
+		return nil, log.LogAndReturnError(kitNetGrpc.ForwardErrorf(codes.InvalidArgument, "cannot delete devices: invalid request"))
+	}
+
+	owner, ownedDevices, err := r.getOwnedDevices(ctx, deviceIds)
```

Comment:
if deviceIds is not you delete all devices at RA but in auth you don't delete any if request doesn't contain deviceIds